### DISCUSSION TITLE
Great algorithm! Using it for a doom port to an eink device and it looks amazing!

### DISCUSSION THREAD

**Discussion by LinusCDE** [mblode/blue-noise#2] [Show and tell]
I found this algorithm a while back. Wanted to port doom to the [reMarkable 1](https://remarkable.com/store/remarkable/). First did a port of doomgeneric to rust and tested it with piston [here](https://github.com/LinusCDE/piston-doom).

At that time the challenge was getting a fast dithering to work. I tested with the piston version (on dither-test) and had horrible (typicial for dithering) performance at first.

Now I really heavily optimized the dithering (using a 320x200px source) and pre-calculating ever possible dither-result, which resulted in a minute loading time and 320MB of ram used. Later I reduced that size to 37,5 by only using bits. Now I also found that it is super compressible and pre-build the cache now during compilation as build script (take about 4s on my machine) and including it in the binary.

In conclusion, the game now runs really fast (11-14FPS on an 1GHz ARMv7 core), looks great and even seems to blow up on an image board, where a friend posted it: https://pr0gramm.com/top/4823217

I still don't understand a lot about dithering and this variant, but it looks really great and can also perform amazing!

Here is the repo and the full version of that early video: github.com/LinusCDE/doomarkable  
I'll likely make another one and polish the game more in a few days.

![image](https://transfer.cosmos-ink.net/3G3F8j/doom_title_screen.jpg)

Kudos for publishing this amazing algorithm in rust, since I'm not sure if this project would have work that great without it!
